paraflow-loader-consumer/src/main/java/cn/edu/ruc/iir/paraflow/loader/consumer/buffer/BufferPool.java

```diff
@@ -57,7 +58,10 @@ public BufferPool(List<TopicFiber> fiberPartitions, long blockCapacity, long flu
 
     public void add(Message message)
     {
-        if (blockSize + message.getValueSize() > blockCapacity) {
+        MessageSizeCalculator messageSizeCalculator = new MessageSizeCalculator();
+        String topic = message.getTopic().get();
+        long messageSize = messageSizeCalculator.caculate(topic);
```

Comment:
Can u change the `calculate()` method into a `static` method ? Or create the `MessageSizeCalculator` object outside `add()` function, because each call is gonna crate a new object, adding gc overhead.